src/components/PokemonSearch.tsx

```diff
@@ -0,0 +1,313 @@
+                    };
+                    setAllPokemon((prevPokemon) => [...prevPokemon, pokemonInfo]);
+                }
+                catch (error) {
+                    console.error("Error fetching pokemon with id ", id, ": ", error);
+                }
+            })
+    
+            await Promise.all(getPokemonPromises);
+        }
+        
+        getPokemonWithIds();
+
+    }, [pokemonIdArray]);
+
+    useEffect(() => {
+        // Get pokemon information using names of the pokemon
+        const getPokemonWithNames = async () => {
+            const getPokemonPromises = pokemonNamesArray.map(async (name) => {
+                try {
+                    const response = await getPokemonByName(name);
+                    let pokemonInfo: PokemonCardInfo = {
+                        name: capitalizeFirstLetter(response.name),
+                        id: response.id,
+                        sprite: response.sprites.other?.["official-artwork"].front_default ?? response.sprites.front_default
+                    };
+                    setAllPokemon((prevPokemon) => [...prevPokemon, pokemonInfo]);
+                } 
+                catch (error) {
+                    console.error("Error fetching pokemon with name, ", name, ": ", error);
+                }
+            });
+
+            await Promise.all(getPokemonPromises);
+        }
+
+        getPokemonWithNames();
+    }, [pokemonNamesArray])
+
+    // Get pokemon information using names of the pokemon
+    const getPokemonInfoWithNames = async (names: string[]) => {
+        const getPokemonPromises = names.map(async (name) => {
+            try {
+                const response = await getPokemonByName(name);
+                let pokemonInfo: PokemonCardInfo = {
+                    name: capitalizeFirstLetter(response.name),
+                    id: response.id,
+                    sprite: response.sprites.other?.["official-artwork"].front_default ?? response.sprites.front_default
+                };
+                setAllPokemon((prevPokemon) => [...prevPokemon, pokemonInfo]);
+            } 
+            catch (error) {
+                console.error("Error fetching pokemon with name, ", name, ": ", error);
+            }
+        });
+
+        await Promise.all(getPokemonPromises);
+    }
+
+    // Get's all the pokemon by type, condenses them, and pulls out the names (for later api call)
+    const getPokemonListByType = async (type: string | null) => {
+        if(type === null) return;
+        try {
+            const response = await getTypeByName(type);
+            const pokemonList = response.pokemon;
+            const num = Math.floor(pokemonList.length/5); // only take 1/5 of array
+            pokemonList.splice(num, pokemonList.length - num);
+            
+            // fetch pokemon
+            let names: string[] = [];
+            pokemonList.map((pokemon) => {
+                names.push(pokemon.pokemon.name);
+            })
```

Comment:
since we're using `map` already, I think we can simplify this to
```
const names = pokemonList.map((pokemon) => {
  return pokemon.pokemon.name; 
})
```